Documentation/guides/D8andR8.md

```diff
@@ -0,0 +1,270 @@
+This is the D8 and R8 integration specification for Xamarin.Android.
+
+# What is D8? What is R8?
+
+At a high level, here are the steps that occur during an Android
+application's Java compilation:
+- `javac` compiles Java code
+- `desugar` remove's the "sugar" (from Java 8 features) that are not
+  fully supported on Android
+- `proguard` shrinks compiled Java code
+- `dx` "dexes" compiled Java code into Android [dex][dex] format. This
+  is an alternate Java bytecode format supported by the Android
+  platform.
+
+This process has a few issues, such as:
+- [proguard](https://www.guardsquare.com/en/products/proguard/manual)
+  is made by a third party, and aimed for Java in general (not Android
+  specific)
+- `dx` is slower than it _could_ be
+
+So in 2017, Google announced a "next-generation" dex compiler named
+[D8](https://android-developers.googleblog.com/2017/08/next-generation-dex-compiler-now-in.html).
+
+- D8 is a direct replacement for `dx`
+- R8 is a replacement for `proguard`, that also "dexes" at the same
+  time. If using R8, a D8 call is not needed.
+
+Both tools have support for various other Android-specifics:
+- Both `desugar` by default unless the `--no-desugaring` switch is
+  specified
+- Both support [multidex][multidex]
+- R8 additionally has support to generate a default `multidex.keep`
+  file, that `proguard` can generate
```

Comment:
I'm not sure what "that `proguard` can generate" is supposed to mean/imply here.